pxe-on-rasppi/README.md

```diff
@@ -0,0 +1,517 @@
+`ldlinux.c32` from the TFTP server running on the Raspberry Pi:
+
+```
+ssh yi@192.168.2.16
+> tftp 192.168.2.10 -c get pxelinux.0
+> tftp 192.168.2.10 -c get ldlinux.c32
+```
+
+### Deploy CoreOS Images
+
+On the Raspberry Pi,
+
+```
+cd /srv/tftp
+CHANNEL=stable
+VERSION=1010.5.0
+wget https://${CHANNEL}.release.core-os.net/amd64-usr/${VERSION}/coreos_production_pxe.vmlinuz
+wget https://${CHANNEL}.release.core-os.net/amd64-usr/${VERSION}/coreos_production_pxe_image.cpio.gz
+```
+
+We can also verify the downloaded image files:
+
+```
+wget https://${CHANNEL}.release.core-os.net/amd64-usr/${VERSION}/coreos_production_pxe.vmlinuz.sig
+wget https://${CHANNEL}.release.core-os.net/amd64-usr/${VERSION}/coreos_production_pxe_image.cpio.gz.sig
+gpg --verify coreos_production_pxe.vmlinuz.sig
+gpg --verify coreos_production_pxe_image.cpio.gz.sig
+```
+
+You might wonder how come the `CHANNEL` and the `VERSION` variables,
+which tells that the most recent stable channel of CoreOS releases the
+1010.5.0 version.  We can always get the up-to-date version of the
+stable channel by accessing
+https://stable.release.core-os.net/amd64-usr/current/version.txt.
+
+## Boot CoreOS
+
+### Configure PXELINUX
+
+The idea that we need pxelinux.0 is that, when a target computer
+boots, it broadcasts a request for IP address.  The DHCP server
+responses an IP address and `next-server` and `filename`.  If the
+network card and BIOS of the target computer supports PXE, it would
+download the boot image from URL `tftp://192.168.1.105/pxelinux.0` and
+keep this URL as *current working directory*.  When it runs
+`pxelinux.0`, it lets it knows the current working directory, so that
+`pxelinux.0` can load its configuration file from there.
+
+Suppose that the MAC address of the network card on the target
+computer is `28-d2-44-fb-19-49`, then `pxelinux.0` will read
+configuration file
+`tftp://192.168.1.105/pxelinux.cfg/01-88-99-aa-bb-cc-dd`.  If that
+file doesn't exists, it tries to read
+`tftp://192.168.1.105/pxelinux.cfg/default`.
+
+The content of our `/var/lib/tftpboot/pxelinux.cfg/default` is as
+follows:
+
+```
+default coreos
+
+label coreos
+  kernel coreos_production_pxe.vmlinuz
+  append initrd=coreos_production_pxe_image.cpio.gz cloud-config-url=http://192.168.1.105:8080/install-coreos.sh
+```
+
+This configuration file tells `pxelinux.0` to download CoreOS images
+`coreos_production_pxe.vmlinuz` and
+`coreos_production_pxe_image.cpio.gz` form its current working
+directory.  Then `pxelinux.0` will boot the system using the CoreOS
+images.
+
+Please be aware that the URL
+`http://192.168.1.105:8080/install-coreos.sh` in above
+cloud-config file doesn't exist at the moment.  We will talk about
+this later.
+
+### Test PXE Booting
+
+Now, restart the Thinkpad and hold Enter to enter the BIOS setup
+interface.  Change the boot order so that Network Boot is on top of
+the list.  Then save and exit.  Now the Thinkpad should boot CoreOS as
```

Comment:
Better to remind the reader, when the computer reboots after the automatic installation, change the boot order and put hard disk in first place. Or, if the BIOS supports, use a one-shot 'boot from pxe' option that doesn't change the boot order int the BIOS setting.